tests/compiler/test_strings.py

```diff
@@ -246,9 +247,14 @@ def test_fstring_relations(snippetcompiler, capsys):
 
 
 def test_fstring_numbering_logic():
+    """
+    Check that variable ranges in f-strings are correctly computed
+    """
     statements = parse_code(
         """
-std::print(f"---{s}{mm} - {sub.attr}")
+std::print(f"---{s}{mm} - {sub.attr} - {  padded  } - {  \tpadded.sub.attr   }")
+#                |   |           |           |                          |
+#                [[ [-[        [---[      [-----[                     [---[     <--- expected ranges
```

Comment:
Cool! Took me a while to interpret it but definitely useful. One potential improvement I think would be to make all ranges exclusive so that it's more visual: the range is whatever is strictly in between the brackets? I would still use `[ ... ]`, even though that clashes with the mathematical notation.